tests/integration/tls_tests/helpers.py

```diff
@@ -70,7 +74,11 @@ async def check_tls(ops_test: OpsTest, unit: ops.model.Unit, enabled: bool) -> b
 
 async def time_file_created(ops_test: OpsTest, unit_name: str, path: str) -> int:
     """Returns the unix timestamp of when a file was created on a specified unit."""
-    time_cmd = f"run --unit {unit_name} --  ls -l --time-style=full-iso {path} "
+    if version("juju") < "3":
```

Comment:
For consistency, I would rather suggest to use `JujuVersion.from_environ().has_secrets`, that providing an overall mock for the test execution. (i.e. in control of Juju version decision on a global scope.)